cypress/support/commands.js

```diff
@@ -87,3 +87,36 @@ Cypress.on('uncaught:exception', (err) => {
     console.log(err);
     return false;
 });
+
+Cypress.Commands.add('checkRequest', ({
+    requestAlias, statusCode,
+}) => {
+    cy
+        .wait(requestAlias)
+        .its('response.statusCode')
+        .should('eq', statusCode);
+});
+
+Cypress.Commands.add('sendRequest', ({
+    requestName, reqType, status,
+}) => {
+    cy
+        .wait(`@${requestName}_${reqType}`)
+        .then((xhr) => {
+            expect(xhr.request.method).to.equal(reqType);
+            expect(xhr.response.statusCode).to.equal(status);
+        });
+});
+
+Cypress.Commands.add('openPage', ({
+    page, requestAlias,
+}) => {
+    cy.visit(`/${page}`);
+    cy
+        .url()
+        .should('include', `/${page}`);
+    cy.checkRequest({
```

Comment:
I would suggest to not mix a context => Open page !== check request. Also after opening page there might be multiple requests going on

Reply:
ok I will edit the `openPage` method to be prepared to accept multiple requests